src/Tree.cpp

```diff
@@ -38,106 +36,101 @@ Tree::
 bool Tree::
 empty()
 {
-  if (cells == nullptr) return true;
+  if (cells_ == nullptr) return true;
   else return false;
 }
 
 void Tree::
 addCell(const Cell* c)
 {
-  cells->push_back(c);
+  cells_->push_back(c);
 }
 
 int Tree::
 countCells()
 {
-  if(cells != nullptr) {
-    // std::cout << "This leaf has " << cells->size() << " elements" << std::endl;
-    return cells->size();
+  if(cells_ != nullptr) {
+    // std::cout << "This leaf has " << cells_->size() << " elements" << std::endl;
+    return cells_->size();
   }
 
   int cellCount = 0;
 
-  cellCount += nw->countCells();
-  cellCount += sw->countCells();
-  cellCount += ne->countCells();
-  cellCount += se->countCells();
+  cellCount += nw_->countCells();
+  cellCount += sw_->countCells();
+  cellCount += ne_->countCells();
+  cellCount += se_->countCells();
 
   return cellCount;
 }
 
-std::vector< const Cell*>* Tree::
+std::unique_ptr<std::vector< const Cell*>> Tree::
 getCells()
 {
-  return cells;
+  return std::move(cells_);
```

Comment:
By moving the cells outside, `cells_` is not valid anymore. Is this what you want to do? See my previous comment on `getCells()`